cds_ils/importer/eitems/importer.py

```diff
@@ -89,7 +93,17 @@ def _update_existing_record(self, existing_eitem, matched_document):
 
     def _delete_existing_record(self, existing_eitem):
         eitem_indexer = current_app_ils.eitem_indexer
-        existing_eitem.delete(force=True)
+        pid = existing_eitem.pid
```

Comment:
Will this work after the deletion? It might be working only because you don't have a session commit yet.
I would probably, to be safe, save in vars:
```
pid_object_type, pid_object_uuid = pid.object_type, pid.object_uuid
```
same below...

Reply:
true, thanks for spotting it! it was working indeed because of the late commit, however in case of changes it is very easy to introduce bugs, I will fix it